index.js

```diff
@@ -26,40 +22,100 @@ var compileCommand = function(p, opts) {
   if (opts.stderr) child.stderr.pipe(opts.stderr)
   else child.stderr.resume()
 
-  var result = duplexify(child.stdin, child.stdout)
+  return duplexify(child.stdin, child.stdout)
+}
 
-  return result
+/*  Create a stream which runs runCommands in sequence */
+var runStream = function (runCommands) {
+  var commands = runCommands.map(function (cmd) {
+    var fn = function () {
+      if (cmd.end) cmd.end() // not writable
+      return cmd
+    }
+    return fn()
+  })
+  return multistream(commands)
 }
 
+/* Run forkCommands in parallel */
+var forkStream = function (forkCommands) {
+  var commands = forkCommands.map(function (cmd) {
+      // no lazyness here since we are forking
+      if (cmd.end) cmd.end() // not writable
+      return cmd
+  })
+  return parallel(commands)
+}
+
+/* Pipe pipeCommands together */
+var pipeStream = function(pipeCommands) {
+  var pipe = pumpify(pipeCommands)
+  pipe.end() // first not writable
+  return pipe
+}
+
+/* Map first command in commands to the rest */
+var mapStream = function(commands) {
+  var pipes = []
+  var first = commands.shift()
+  commands.forEach(function(mapped) {
+    pipes.push(pipeStream([first, mapped]))
+  })
+  return forkStream(pipes)
+}
+
+/* Map the rest of the commands to the first command */
+var reduceStream = function(commands) {
```

Comment:
Yeah, I was considering that!  I left it like this because I thought it might be more readable, but I don't think abstracting away makes it that confusing.